src/components/extendedPopover/ExtendedPopover.tsx

```diff
@@ -0,0 +1,163 @@
+import * as React from "react";
+import * as ReactDOM from "react-dom";
+import styled from "styled-components";
+import { useHideOnScroll, useHandleClickOutside } from "../../hooks";
+import theme from "../../theme";
+import {
+  Placement,
+  Position,
+  getStyle,
+  Margin,
+  DEFAULT_MARGIN,
+  PaddingWrapper,
+} from "../../utils/position";
+import { Button } from "../button";
+import { SVGIcon } from "../svgicon";
+
+type Props = {
+  content: React.ReactNode;
+  trigger?: React.ReactNode;
+  placement?: Placement;
+  position?: Position;
+  margin?: Margin;
+  clickOutsideToClose?: boolean;
+  triggerOn?: ExtendedPopoverAction;
+  closeOn?: ExtendedPopoverAction;
+};
+
+export function ExtendedPopover({
+  trigger = (
+    <Button size="sm">
+      <SVGIcon size="sm" name="burgerMenu" />
+    </Button>
+  ),
+  content,
+  placement = "right",
+  position = "start",
+  clickOutsideToClose = true,
+  margin = DEFAULT_MARGIN,
+  triggerOn = "click",
+  closeOn = "click",
+}: Props) {
+  const triggerRef = React.useRef<HTMLDivElement>(null);
+  const contentRef = React.useRef<HTMLDivElement>(null);
+
+  const [open, setOpen] = React.useState(false);
+  const [contentDimensions, setContentDimensions] =
+    React.useState<DOMRect | null>(null);
+  const [triggerDimensions, setTriggerDimensions] =
+    React.useState<DOMRect | null>(null);
+
+  React.useEffect(() => {
+    if (
+      open &&
+      contentRef.current instanceof HTMLDivElement &&
+      triggerRef.current instanceof HTMLDivElement
+    ) {
+      setTriggerDimensions(triggerRef.current.getBoundingClientRect());
+      setContentDimensions(contentRef.current.getBoundingClientRect());
+    }
+  }, [open]);
+
+  const handleClickOutside = React.useCallback(
+    (event) => {
+      if (
+        contentRef.current &&
+        triggerRef.current &&
+        !contentRef.current.contains(event.target) &&
+        !triggerRef.current.contains(event.target)
+      ) {
+        setOpen(false);
+      }
+    },
+    [contentRef, triggerRef]
+  );
+
+  useHandleClickOutside({ handleClickOutside, clickOutsideToClose });
+
+  const handleScroll = React.useCallback(() => {
+    setOpen(false);
+  }, []);
+
+  useHideOnScroll({ handleScroll });
+
+  if (content == null) {
+    return null;
+  }
+
+  const triggerProps = {
+    ...(triggerOn === "click"
+      ? {
+          onClick: () => setOpen(!open),
```

Comment:
better be safe and use function update
```suggestion
          onClick: () => setOpen(prevState => !prevState),
```